src/box/memtx_tx.c

```diff
@@ -1253,6 +1253,20 @@ memtx_tx_history_add_stmt(struct txn_stmt *stmt, struct tuple *old_tuple,
 void
 memtx_tx_history_rollback_stmt(struct txn_stmt *stmt)
 {
+	if (stmt->del_story != NULL) {
```

Comment:
Mb it is worth moving this part to a separate static-helper? It looks like independent chunk of code.